cfs_config_util/cfs.py

```diff
@@ -414,15 +414,19 @@ def save_to_cfs(self, name=None, overwrite=True):
         if not overwrite:
             try:
                 response = self._cfs_client.get('v2', 'configurations', name, raise_not_ok=False)
-                if response.status_code != 404:
+
+                # If response was OK, that indicates there's already a CFS configuration
+                if response.ok:
                     raise CFSConfigurationError(f'A configuration named {name} already exists '
-                                                f'and may not be overwritten.')
+                                                f'and will not be overwritten.')
+                elif response.status_code != 404:
+                    # If there's a failure for any reason other than a missing
+                    # configuration, throw an APIError
+                    self._cfs_client.raise_from_response(response)
```

Comment:
I like the idea behind this change which appears to be intended to provide consistency with how the non-404 errors are represented, but I think we need to do something a little different given that `raise_from_response` doesn't have the `req_type` and `url` variables defined in that context. We could probably access the `response` object's `request.url` and `request.method` to get at those, but testing is needed to be sure that gets us what we want.

https://requests.readthedocs.io/en/latest/api/#requests.Response.request

Reply:
> I like the idea behind this change which appears to be intended to provide consistency with how the non-404 errors are represented, but I think we need to do something a little different given that `raise_from_response` doesn't have the `req_type` and `url` variables defined in that context. We could probably access the `response` object's `request.url` and `request.method` to get at those, but testing is needed to be sure that gets us what we want.

I tested this this morning and it seems to get us what we want. (There aren't any unit tests for the API client in this repo yet, though I can add them if you'd like.) I've pushed a new fixup commit, fbec7d4, to address this.